components/eam/src/physics/cam/modal_aer_opt.F90

```diff
@@ -65,6 +66,17 @@ module modal_aer_opt
 character(len=4) :: diag(0:n_diag) = (/'    ','_d1 ','_d2 ','_d3 ','_d4 ','_d5 ', &
                                        '_d6 ','_d7 ','_d8 ','_d9 ','_d10'/)
 
+!Declare the following threadprivate variables to be used for calcsize and water uptake
+!These are defined as module level variables to aviod allocation-deallocation in a loop
+real(r8), pointer :: dgnumdry_m(:,:,:) ! number mode dry diameter for all modes
+real(r8), pointer :: dgnumwet_m(:,:,:) ! number mode wet diameter for all modes
+real(r8), pointer :: qaerwat_m(:,:,:)  ! aerosol water (g/g) for all modes
+real(r8), pointer :: wetdens_m(:,:,:)  ! aerosol water (g/g) for all modes
+!$OMP THREADPRIVATE(dgnumdry_m, dgnumwet_m, qaerwat_m)
```

Comment:
@singhbalwinder should wetdens_m appear in the threadprivate list?

Reply:
`wetdens` is not being used in the code (or impact any calculations) and I have removed it entirely in my revised code (PR pending). 

I created this PR (with minimal changes)so that we can get all the NBFB changes in the master. Sorry about the confusion.